File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& ht) {
        int mx_water{}, i{}, j=ht.size()-1;
        
       while(i<j){
           int h=min(ht[i],ht[j]);
           mx_water=max(mx_water, (j-i)*h);
           
           while(i<j && ht[i]<=h) ++i;
           while(i<j && ht[j]<=h) --j;
       }
        return mx_water;
    }
};

File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] ht) {
        int mx_water=0, l=0, r=ht.length-1;
        
        while(l<r){
            int mn_ht=Math.min(ht[l], ht[r]);
            mx_water= Math.max(mx_water,(r-l)*mn_ht);
            
            while(l<r && ht[l]<=mn_ht) ++l;
            while(l<r && ht[r]<=mn_ht) --r;
        }
        return mx_water;
    }
}